autocertkit/test_runner.py

```diff
@@ -174,6 +174,69 @@ def update_xml_with_result(dom, class_node, results):
 
         recurse_add_records_to_node(method_node, result)
 
+def get_failed_cases(test_file):
```

Comment:
Excessive duplicate codes need to optimize.
As function is the first class citizen in python, you can pass it as parameter.
something like as follows:
```
def get_cases(test_file, func_name):
    tm_cases = []
    devices = models.create_models(test_file)
    
    for device in devices:
        tm_cases += getattr(device, func_name)()
    
    return tm_cases
```
usage
```
tm_failed = get_cases(test_file, "get_failed_test_methods")
tm_rerun = get_cases(test_file, "get_rerun_test_methods")
```

Reply:
Nice!